pkg/kubelet/cm/deviceplugin/device_plugin_handler.go

```diff
@@ -337,19 +344,21 @@ func (h *HandlerImpl) writeCheckpoint() error {
 	if err != nil {
 		return err
 	}
-	filepath := h.devicePluginManager.CheckpointFile()
-	return ioutil.WriteFile(filepath, dataJSON, 0644)
+	err = h.store.Write(defaultCheckpoint, dataJSON)
+	if err != nil {
+		return fmt.Errorf("failed to write checkpoint file %q: %v", defaultCheckpoint, err)
+	}
+	return nil
 }
 
 // Reads device to container allocation information from disk, and populates
 // h.allocatedDevices accordingly.
 func (h *HandlerImpl) readCheckpoint() error {
-	filepath := h.devicePluginManager.CheckpointFile()
-	content, err := ioutil.ReadFile(filepath)
-	if err != nil && !os.IsNotExist(err) {
-		return fmt.Errorf("failed to read checkpoint file %q: %v", filepath, err)
+	content, err := h.store.Read(defaultCheckpoint)
```

Comment:
What error does `Read` return when the file is not found? When the kubelet is started for the first time, the checkpoint file won't exist but we don't want to error during `readCheckpoint()` in that case.